bundle/src/main/java/com/adobe/acs/commons/mcp/util/CompositeVariant.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2018 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.mcp.util;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a value which could be either a list of variants or a single variant. The idea is that this supports a
+ * transition from a singular to a multi-value property for easier conversion.
+ *
+ * @param <T> Can be of any class supported by Variant, used for direct conversion in getValue, etc.
+ */
+public class CompositeVariant<T> {
```

Comment:
should be final and/or @ProviderType

Reply:
and actually... does this need to be part of the API? Seems like it is only used internally.